### PR TITLE
Add ADR proposal for EKS

### PR DESCRIPTION
**Pull request by cmharlow**
Jira: https://jira.mozilla.com/browse/SE-1673

What this PR does: 
* Includes an ADR proposal for using EKS & Kubernetes for Web SRE-managed services
* Also includes work in #3 as part of keeping TOC up to date, so ignore the Service Doc ADR until its merged into master & this is rebased.

Notes:
* This is intentionally kept pretty minimal (though I'm sure we could pare it down more). 
* This obviously changes when GCP becomes available to us; its meant to give us an interim baseline however

### CODE REVIEW

**Review by floatingatoll — APPROVED** [review]
No objection to 0003-use-eks, though I would prefer at least one other reviewer signoff.

File: decisions/websre/0003-use-eks-cluster-as-default-container-platform.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+* We will use AWS' Elastic Kubernetes Service to manage our Kubernetes clusters, unless there is a documented reason for a particular project not to.
+* We will deploy our EKS clusters using Terraform & our internal Terraform module, unless there is a documented reason for a particular project not to.
+* Any divergences from the above decisions must be clearly documented in the project's or application's service documentation.
+
+## Consequences
+
+* Reduces complexity of setting up our own infrastructure, or context switching between multiple container platform infrastructures.
+* Incurs EKS costs.
+* Will require migrations from old setups (Kops, External-terraform EKS module).
+
```
Comment (duallain) [inline decisions/websre/0003-use-eks-cluster-as-default-container-platform.md:30]: A really good consequence of this choice is:

* EKS access model should improve our security posture and ease of access.
  * Users must have SSO access to get to a cluster, so employees exitng limits access more reliably than previous k8s models.
  * Having the correct AWS permissions allows you to create the k8s credentials needed to interact with the cluster, so we no longer have to pass a secret file around between people

**Review by smarnach — APPROVED** [review]
Looking good to me. I fully agree with this decision as our go-to solution until we have standardized how we deploy to GCP across SRE teams.

File: decisions/websre/0002-web-sre-service-documentation.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# 2. Web SRE Service Documentation
+
+Date: 2021-06-01
+Scope: Web SRE Team
+
+## Status
+
+Proposed
```
Comment (smarnach) [inline decisions/websre/0002-web-sre-service-documentation.md]: Will we change this to "Accepted" before merging? I guess we don't really need to record the status in the ADR – if the PR is open, it's "proposed". If the PR gets closed, it's "rejected" and if we merge it, it's "accepted".

File: decisions/websre/0003-use-eks-cluster-as-default-container-platform.md (hunk 1 above)
Comment (cmharlow) [reply decisions/websre/0003-use-eks-cluster-as-default-container-platform.md:30]: will add, thanks for writing up

Comment (cmharlow) [reply decisions/websre/0003-use-eks-cluster-as-default-container-platform.md:30]: added, let me know if it looks okay @duallain

File: decisions/websre/0002-web-sre-service-documentation.md (hunk 2 above)
Comment (cmharlow) [reply decisions/websre/0002-web-sre-service-documentation.md]: I've just been adding a commit saynig it was accepted pre PR merge. Open to other ways though

File: decisions/websre/0003-use-eks-cluster-as-default-container-platform.md (hunk 1 above)
Comment (duallain) [reply decisions/websre/0003-use-eks-cluster-as-default-container-platform.md:30]: LGTM, thanks.  Will resolve conversation.

**Review by duallain — APPROVED** [review]

**Review by sciurus — APPROVED** [review]

**Pull request merged**